substrate/frame/support/src/dispatch.rs

```diff
@@ -579,6 +579,14 @@ impl<T> ClassifyDispatch<T> for (Weight, DispatchClass, Pays) {
 	}
 }
 
+impl AccrueWeight for PostDispatchInfo {
+	fn accrue(&mut self, weight: Weight) {
+		if let Some(actual_weight) = self.actual_weight.as_mut() {
+			*actual_weight = actual_weight.saturating_add(weight);
+		}
+	}
```

Comment:
this is not ideal because if the dispatch call returns no refund, then the transaction extensions cannot refund (their post dispatch details will accumulate into `None`).

But it should be fixed by the redesign into decreasing post dispatch info weight.